### PR TITLE
tidy up VSCode lookup

### LINKED ISSUE #3304: Desktop can't detect latest VS Code on Win7 x64 (+workaround)
**Problem:** Windows 7 (x64), fresh install of latest Github Desktop (1.0.8) and Visual Studio Code (1.17+), GD can't detect VSC as editor.

**Causes:** After figuring out how `win32.ts` detects the installed editors and watching the verbose log in dev tools during detection, I found two possible causes:

1. Lately VSC changed its `displayName` key to *"Microsoft Visual Studio Code"*.
2. If you installed VSC to a different Start Menu group (*"utils\VSC"* in my case), detection will also fail because of the non-default `Inno Setup: Icon Group` key value.

**Workaround:** Change both `displayName` and `Inno Setup: Icon Group` keys to *"Visual Studio Code"*. These are under *HKLM\SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\\{EA457B21-F73E-494C-ACAB-524FDE069978}_is1* on Win7-x64. Re-run Github Desktop detection (File - Options). It should detect VSC correctly now and VSC should appear in the Repository menu. Save the config, then change back both registry keys to their original values. If you ever re-open Options, detection will fail again, so you have to do this every time you change Desktop preferences (until GD will get a better detection).

### PR DESCRIPTION
**Pull request by shiftkey**
Fixes #3304 and adds some extra context.

The original use of `Inno Setup: Icon Group` was because Sublime Text bakes the build number into it's `DisplayName` value. That's going to break whenever they release a new update.

VSCode however doesn't do this, so I broke this out to it's own code path and switched back to `DisplayName` because that's what everyone expects.

And the Insiders build is done in the same way:

### CODE REVIEW

**Comment by shiftkey** [comment]
Also going to tweak the install location for both of these tools to avoid more `Inno Setup` confusion. `InstallLocation` is accurate for both:

File: app/src/lib/editors/win32.ts [hunk 1]
```diff
@@ -159,11 +157,26 @@ function extractApplicationInformation(
...
+    return { displayName, publisher, installLocation }
+  }
+
+  if (editor === ExternalEditor.SublimeText) {
     for (const item of keys) {
+      // NOTE:
+      // Sublime Text bakes the build number into the DisplayName value, so for
+      // forward-compatibility whenever they decide to drop a new build let's
+      // use this entry which doesn't contain the version number
       if (item.name === 'Inno Setup: Icon Group') {
```
Comment (niik) [inline app/src/lib/editors/win32.ts]: I have absolutely no clue about how Inno Setup but https://github.com/desktop/desktop/issues/3304#issuecomment-344340564 mentions that Icon Group could be affected by user input which seems bad. That comment was made about VS Code but if that applies to Sublime Text as well could we use `displayName` but check for a prefix match instead? I.e.

```ts
else if (item.name === 'DisplayName' && /^Sublime Text/.test(item.value)) {
    displayName = 'Sublime Text'
}

Comment (joshaber) [reply app/src/lib/editors/win32.ts]: Or `String.prototype.startsWith()`?

Comment (niik) [reply app/src/lib/editors/win32.ts]: > Or `String.prototype.startsWith()`?

Yeah, I just didn't know if `item.value` could be null/undefined. The RegEx route would work regardless of what we pass it whereas `startsWith` _could_ need an additional check. I'm down for either or.

```ts
else if (item.name === 'DisplayName' && item.value && item.value.startsWith('Sublime Text')) {
    displayName = 'Sublime Text'
}
```

File: app/src/lib/editors/win32.ts [hunk 2]
```diff
@@ -111,9 +110,8 @@ function isExpectedInstallation(
       return displayName === 'Atom' && publisher === 'GitHub Inc.'
     case ExternalEditor.VisualStudioCode:
       return (
-        (displayName === 'Visual Studio Code' ||
```
[the remark this answers is not in the document]
Comment (shiftkey) [reply app/src/lib/editors/win32.ts]: This is the value stored in `Inno Setup: Icon Group`, which we no longer read for VSCode.

File: app/src/lib/editors/win32.ts (hunk 1 above)
Comment (shiftkey) [reply app/src/lib/editors/win32.ts]: Yeah, I'll suck it up and be more tolerant of the Sublime Text values 👍

**Comment by shiftkey** [comment]
:apple:

**Pull request merged**